### PR TITLE
[WIP] boards name and number external SPI consistently

### PR DESCRIPTION
**Pull request by dagar**
This PR unifies the mess of different external SPI bus naming and numbering across boards.

**Master**

 * PX4_SPI_BUS_EXT
 * PX4_SPI_BUS_EXT0
 * PX4_SPI_BUS_EXT1
 * PX4_SPI_BUS_EXTERNAL1
 * PX4_SPI_BUS_EXTERNAL2
 * PX4_SPI_BUS_EXPANSION

**PR**

 * PX4_SPI_BUS_EXTERNAL**x** (starts at 1)

Gradually working towards a simple unified mechanism for handling optional external peripherals without pushing the burden into every single driver and board.

### CODE REVIEW

File: src/drivers/boards/common/board_common.h [hunk 1]
```diff
@@ -1067,11 +1067,11 @@ __EXPORT bool px4_spi_bus_external(int bus);
 
 #else
 
-#ifdef PX4_SPI_BUS_EXT
-#define px4_spi_bus_external(bus) (bus == PX4_SPI_BUS_EXT)
+#ifdef PX4_SPI_BUS_EXTERNAL1
+#define px4_spi_bus_external(bus) (bus == PX4_SPI_BUS_EXTERNAL1)
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/drivers/boards/common/board_common.h]: Yes

File: boards/nxp/fmuk66-v3/src/board_config.h [hunk 2]
```diff
@@ -244,17 +243,17 @@ __BEGIN_DECLS
...
 #define PX4_EXTERNAL_BUS_CS_GPIO            {GPIO_SPI2_CS, GPIO_SPI2_EXT}
-#define PX4_EXTERNAL_BUS_FIRST_CS           PX4_SPIDEV_EXTERNAL1
-#define PX4_EXTERNAL_BUS_LAST_CS            PX4_SPIDEV_EXTERNAL2
-
-#define PX4_SPIDEV_ICM_20602                PX4_SPIDEV_EXTERNAL1
-#define PX4_SPIDEV_ICM_20608                PX4_SPIDEV_EXTERNAL1
-#define PX4_SPIDEV_ICM_20689                PX4_SPIDEV_EXTERNAL1
-#define PX4_SPIDEV_EXT_MPU                  PX4_SPIDEV_EXTERNAL1
-#define PX4_SPIDEV_MPU                      PX4_SPIDEV_EXTERNAL1
+#define PX4_EXTERNAL_BUS_FIRST_CS           PX4_SPIDEV_EXTERNAL1_1
```
[the remark this answers is not in the document]
Comment (dagar) [reply boards/nxp/fmuk66-v3/src/board_config.h]: Actually, I don't think we even need them. Update coming.

[the remark this answers is not in the document]
Comment (dagar) [reply boards/nxp/fmuk66-v3/src/board_config.h]: How about this instead? https://github.com/dagar/Firmware/blob/pr-fmu-v5_bsp_spi/boards/px4/fmu-v5/src/spi.cpp#L121-L132

[the remark this answers is not in the document]
Comment (davids5) [reply boards/nxp/fmuk66-v3/src/board_config.h]: Wicked! - and scarry - I like it!

[the remark this answers is not in the document]
Comment (dagar) [reply boards/nxp/fmuk66-v3/src/board_config.h]: https://github.com/PX4/Firmware/pull/11370

**Comment by julianoes** [comment]
@davids5 what's up with this?

**Comment by davids5** [comment]
@julianoes - it was @dagar work and has rotted. If if cleans it up and resolve the open, we can review it again. 

I would make the CS as external in the ID - i.e PX4_MK_SPI_EXT_SEL - then they are self describing.

**Pull request closed without merging**